tt_eager/tt_dnn/op_library/concat/kernels/dataflow/reader_concat_stick_layout_interleaved_start_id.cpp

```diff
@@ -32,7 +32,7 @@ void kernel_main() {
     constexpr uint32_t num_pages_per_block_base_offset = is_dram_base_offset + num_tensors;
     constexpr uint32_t page_size_per_tensor_offset = num_pages_per_block_base_offset + num_tensors;
     constexpr uint32_t page_id_per_tensor_offset = page_size_per_tensor_offset + num_tensors;
-    volatile tt_l1_ptr uint32_t * arg_ptr = (volatile tt_l1_ptr uint32_t*) get_arg_addr(src_addr_base_idx);
+    tt_l1_ptr uint32_t * arg_ptr = (tt_l1_ptr uint32_t*) get_arg_addr(src_addr_base_idx);
```

Comment:
Are these changes to ensure that the compiler doesn't redo address calculations?

Reply:
I was looking at this and wondering why we use volatile so much.  There is nothing volatile about these values during the lifetime of a kernel.  They are loaded before the kernel runs and can only be changed if the kernel writes to them.  The only odd thing is that that the address isn't "known" by the compiler (ie, not a compiler variable), but we have a pointer pointing to a range and the pointer comes from the launch message and the compiler has no idea.  So they don't need to be volatile.  And...yeah, maybe that will help generate better code (volatile definitely perturbs code generation).